sktime/tests/test_all_estimators.py

```diff
@@ -49,6 +49,48 @@
 )
 
 
+def _create_all_test_instances(cls):
+    """Create list of all test instances.
+
+    Parameters
+    ----------
+    cls : sktime class inheriting from BaseObject
+
+    Returns
+    -------
+    objs : list of instances of cls
+        i-th instance is cls(**cls.get_test_params()[i])
+    names : list of str, same length as objs
+        i-th element is name of i-th instance of obj in tests
+        convention is {cls.__name__}-{i} if more than one instance
+        otherwise {cls.__name__}
+    """
+    objs = []
+    param_list = cls.get_test_params()
+    if not isinstance(param_list, (dict, list)):
+        raise RuntimeError(
+            f"Error in {cls.__name__}.get_test_params, "
+            "return must be param dict for class, or list thereof"
+        )
+    if isinstance(param_list, dict):
+        param_list = [param_list]
+    for params in param_list:
+        if not isinstance(params, dict):
+            raise RuntimeError(
+                f"Error in {cls.__name__}.get_test_params, "
+                "return must be param dict for class, or list thereof"
+            )
+        objs += [cls(**params)]
+
+        num_instances = len(param_list)
+        if num_instances > 1:
+            names = [cls.__name__ + "-" + str(i) for i in range(num_instances)]
+        else:
+            names = [cls.__name__]
```

Comment:
yes, correct, this was a mistake, thanks!
It did not have an effect on correctness though since each execution created the same variable assignment, including the last one which would be identical with the variable assignment of the "correct" code.

Fixed.